src/main/java/com/rakovets/course/java/core/practice/io/file_analyzer/FileAnalyzeUtil.java

```diff
@@ -0,0 +1,151 @@
+    public static Map<String, Integer> takeFilePathGetFrequencyRepetitionAllWordsInTextAscendingOrder(Path filePath) {
+        Map<String, Integer> repetitionOfWords = new HashMap<>();
+        String[] words = takeFilePathGetListString(filePath).toString().trim()
+                .replaceAll("[\\[\\],.!?]", "").split(" ");
+        for (String word : words) {
+            repetitionOfWords.put(word, repetitionOfWords.getOrDefault(word, 0) + 1);
+        }
+        return repetitionOfWords.entrySet().stream()
+                .sorted(Map.Entry.comparingByValue())
+                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (e1, e2) -> e2, LinkedHashMap::new));
+    }
+
+    //task8
+    public static void takeFilePathGetSortedNumbersInFile(Path filePath) {
+        String[] arrayNumbers = takeFilePathGetListString(filePath).toString()
+                .replaceAll("[^0-9\\s]", "").split(" ");
+        List<Integer> sortedNumbers = Arrays.stream(arrayNumbers)
+                .map(Integer::parseInt)
+                .sorted()
+                .collect(Collectors.toList());
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(filePath + "_"))) {
+            bufferedWriter.write(sortedNumbers.toString());
+            bufferedWriter.flush();
+        } catch (IOException ex) {
+            System.out.println(ex.getMessage());
+        }
+    }
+
+    //task9
+    public static double takeFilePathGetAcademicPerformanceOfStudents(Path filePath) {
+        String text;
+        double countMarks = 0;
+        double sumMarks = 0;
+        double averageMark;
+        try (BufferedReader reader = new BufferedReader(new FileReader(String.valueOf(filePath)))) {
+            while ((text = reader.readLine()) != null) {
+                String[] array = text.trim().replaceAll("[a-zA-Z]", "").split("[, ]");
+                for (String value : array) {
+                    try {
+                        sumMarks += Double.parseDouble(value.trim());
+                        countMarks++;
+                    } catch (NumberFormatException ignored) {
+                    }
+                }
+            }
+        } catch (IOException ex) {
+            System.out.println(ex.getMessage());
+        }
+        averageMark = BigDecimal.valueOf(sumMarks / countMarks).setScale(2, RoundingMode.HALF_UP).doubleValue();
+        return averageMark;
+    }
+
+    //task10
+    public static void takeFilePathGetReplaceModifier(String filePath, String oldModifier, String newModifier) {
+        String newFilePath = filePath.replace(filePath, filePath + "_");
+        String javaCode;
+        String newJavaCode;
+        try (BufferedReader reader = new BufferedReader(new FileReader(filePath));
+             BufferedWriter writer = new BufferedWriter(new FileWriter(newFilePath))) {
+            while ((javaCode = reader.readLine()) != null) {
+                newJavaCode = javaCode.replace(oldModifier, newModifier);
```

Comment:
This isn't correct because it will rename method `privateMyNameMethod` to `publicMyNameMethod`